src/System.Private.CoreLib/shared/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -463,6 +463,33 @@ private void HandleSatelliteMissing()
             throw new MissingSatelliteAssemblyException(SR.Format(SR.MissingSatelliteAssembly_Culture_Name, _mediator.NeutralResourcesCulture, satAssemName), missingCultureName);
         }
 
+        private static string GetManifestResourceNamesList(Assembly assembly)
+        {
+            try
+            {
+                string [] manifestResourceNames = assembly.GetManifestResourceNames();
+                if (manifestResourceNames.Length == 0)
+                {
+                    return "[]";
+                }
+
+                StringBuilder sb = StringBuilderCache.Acquire();
+                sb.Append('[');
```

Comment:
Use `string.Join` ?

Reply:
string.Join would be preferred here? wouldn't it create more string objects?

Reply:
This is error path. Performance does not matter. The small concise code is better.

Reply:
I addressed that.